### PR TITLE
Fix for Unavailability of Supabase API Key and URL at Build Time

### PR DESCRIPTION
**Pull request by okikio**
Title: **Async Environment Variables Implementation for Supabase API Key and URL**

Description:
### Summary

This pull request introduces a change to the way we handle our environment variables, specifically the `PUBLIC_SUPABASE_API_KEY` and `PUBLIC_SUPABASE_API_URL`. Previously, these variables were handled as static values, accessed at build time. This PR modifies our approach to fetch these values asynchronously, thereby allowing us to access them at runtime.

### Problem

The previous implementation encountered issues with accessing the `PUBLIC_SUPABASE_API_KEY` and `PUBLIC_SUPABASE_API_URL` at build time. Due to restrictions within our current infrastructure, these values were not available, which limited our ability to interact with Supabase services during the build phase.

### Solution

To overcome this, I've rewritten our approach to make the environment variables asynchronous. This change means that the Supabase environment variables are fetched when the application is running, rather than at build time.

### Implementation Details

1. **Fetching Env Vars at Runtime**: Created an async function that fetches the `PUBLIC_SUPABASE_API_KEY` and `PUBLIC_SUPABASE_API_URL`. This function will be run when the application starts.

2. **Using the Env Vars**: The Supabase API key and URL are fetched via the aforementioned function when needed in the application. This is handled in an asynchronous context to manage the promises returned.

3. **Building without the Env Vars**: During the Astro application build, the Supabase environment variables will not be available. This is a known limitation, and components have been updated to handle this gracefully (e.g., loading states or fallbacks).

4. **Applying the Env Vars at Runtime**: When the built Astro application is run, the async function will fetch and apply the Supabase environment variables. Components have been designed to reactively update when these values are available.

This change allows for better flexibility in our application, as we are no longer limited by the availability of the Supabase environment variables at build time. 

Please, review and provide any feedback you might have.

### CODE REVIEW

File: packages/astro/.astro/types.d.ts [hunk 1]
```diff
@@ -65,53 +64,147 @@ declare module 'astro:content' {
...
+					id: keyof DataEntryMap[C];
+			  }
+	>;
+	// Allow generic `string` to avoid excessive type errors in the config
+	// if `dev` is not running to update as you edit.
+	// Invalid collection names will be caught at build time.
+	export function reference<C extends string>(
+		collection: C
+	): import('astro/zod').ZodEffects<import('astro/zod').ZodString, never>;
+
```
Comment (aFuzzyBear) [inline packages/astro/.astro/types.d.ts:184]: I take it this is to get Zod to conform to our own use cases or is this a problem upstream with Astro/Zod?

File: packages/astro/src/db/ssr.ts [hunk 2]
```diff
@@ -1,42 +1,98 @@
...
-        ...options,
-        httpOnly: true,
-      });
-    },
-    getRequestHeader: (key) => {
-      return context.request.headers.get(key) ?? undefined;
-    },
-    cookieOptions,
-  });
+export class AstroCookieAuthStorageAdapter extends CookieAuthStorageAdapter {
```
Comment (aFuzzyBear) [inline packages/astro/src/db/ssr.ts:21]: noice

**Review by aFuzzyBear — APPROVED** [review]
This is a fairly sugrical fix for getting Supabase working with Astro in buildtime.
Great work @okikio really did enjoy reviewing this one

**Comment by aFuzzyBear** [comment]
There is a parse error within `packages/astro/src/utils/consts.ts` can you take a look at that for us @okikio ?

**Comment by okikio** [comment]
When running the script using the vercel cli that error doesn't occur, I'm so confused, but I think I've finally figured out the issue. Vercel Serverless doesn't like `import.meta.env` for some reason.

Here is the log https://vercel.com/mthrshp/skf/8TRXzP1Pm4pGzVpjy7xELagXdu3m#L25
CleanShot 2023-07-19 at 19 43 13@2x

**Comment by okikio** [comment]
I just set it up to use `process.env` and that fixed the problem, this is ready for review

File: k8s/.docker/docker-compose.yml [hunk 3]
```diff
@@ -32,7 +32,7 @@ services:
     environment:
       - SKF_LABS_DOMAIN=http://localhost
       - SKF_LABS_DEPLOY_MODE=port
-      - PUBLIC_SUPABASE_API_KEY=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImNjc2dmb29hbmtja2ZxcG1jZnliIiwicm9sZSI6ImFub24iLCJpYXQiOjE2NjkwNTYzNDcsImV4cCI6MTk4NDYzMjM0N30.aicU94g4B2IThg69VEhScf37hqJTzOc-FXMvTEX-br4
```
Comment (aFuzzyBear) [inline k8s/.docker/docker-compose.yml:35]: 👀 great catch

File: packages/astro/package.json
Comment (aFuzzyBear) [inline packages/astro/package.json:1]: Good bump

File: packages/astro/src/db/client.ts
Comment (aFuzzyBear) [inline packages/astro/src/db/client.ts:1]: This is impressive.

**Review by aFuzzyBear — APPROVED** [review]
This is a very intricated fix for supabase. 
💚

**Comment by aFuzzyBear** [comment]
!!! LGTM !!!

**Pull request merged**